web/hooks/push.go

```diff
@@ -0,0 +1,259 @@
+			return
+		}
+		wh.updateLastActivityDate(jobOwner.ID, course.ID)
+		assignments := wh.extractAssignments(payload, course)
+		for _, assignment := range assignments {
+			if assignment.IsGroupLab {
+				// only run group assignments
+				// TODO(Espeland): Need to find a way to have these tests run on the correct branch
+				results := wh.runAssignmentTests(assignment, repo, course, payload)
+
+				if !isDefaultBranch(payload) {
+					// Attempt to find the pull request for the branch, if it exists,
+					// and then assign reviewers to it, if the branch task score is higher than the assignment score limit.
+					wh.handleTaskPush(payload, results, assignment, course, repo)
+				}
+			} else {
+				wh.logger.Debugf("Ignoring assignment: %s, pushed to group repo: %s", assignment.GetName(), payload.GetRepo().GetName())
+			}
+		}
+
+	default:
+		wh.logger.Debug("Nothing to do for this push event")
+	}
+}
+
+// handleTaskPush attempts to find a task for the pushed to branch.
+// It then extracts the score from all tests belonging to that task
+// If a passing score is reached, reviewers are assigned.
+func (wh GitHubWebHook) handleTaskPush(payload *github.PushEvent, results *score.Results, assignment *pb.Assignment, course *pb.Course, repo *pb.Repository) {
+	wh.logger.Debugf("Attempting to find pull request for ref: %s, in repository: %s",
+		payload.GetRef(), payload.GetRepo().GetFullName())
+
+	pullRequest, err := wh.db.GetPullRequest(&pb.PullRequest{
+		SourceBranchName:     getBranchName(payload.GetRef()),
+		ExternalRepositoryID: uint64(payload.GetRepo().GetID()),
+	})
+	if err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			wh.logger.Debugf("No pull request found for ref: %s", payload.GetRef())
+			return
+		}
+		wh.logger.Errorf("Failed to get pull request from database: %v", err)
+		return
+	}
+	tasks, err := wh.db.GetTasks(&pb.Task{ID: pullRequest.GetTaskID()})
+	if err != nil {
+		// A pull request should always have a task association, we therefore log an error.
+		wh.logger.Errorf("Failed to get task from the database: %v", err)
+		return
+	}
+	if len(tasks) != 1 {
+		// This should never happen
+		wh.logger.Errorf("Got an unexpected number of tasks: %d", len(tasks))
+		return
+	}
+	task := tasks[0]
+	// TODO(Meling): My idea is that when a teacher wants to assign a test to a specific task they will use the 'local' task name.
+	// For example, if a teacher has created the markdown file task-hello_world.md,
+	// they would do scores.AddWithTaskName(TestHelloWorld, "task-hello_world", max, weight).
+	// For the proceeding logic we need the 'global' task name however, i.e. if task-hello_world.md is in assignment1,
+	// then the full task name would be assignment1/hello_world.
+	// Do you concur with this approach?
+	// We could of course simply make teachers assign the global task name, but that seems somewhat counter-intuitive to me.
```

Comment:
While reviewing this I imagined a situation where a group pushes to two group assignments at the same time, and where the teacher has created tasks in both with the same name. E.g. groupAssignment1/task-1 and groupAssignment2/task-1. QF would detect changes in both the assignments, but given that it only scores on a per assignment basis, I think it would still work for teachers to use local task names in both.